src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java

```diff
@@ -843,17 +855,27 @@ private static void logQuery(DatastoreMetricQuery query,
                                                                     List<String> useSplit,
                                                                     int limit) {
         final List<DataPointsRowKey> rowKeys = new LinkedList<>();
-        collectFromRowKeySplitIndex(rowKeys,
-                query,
-                useSplitField,
-                useSplit,
-                query.getStartTime(),
-                query.getEndTime(),
-                limit);
+        if (m_cassandraConfiguration.isUseTimeIndexRead()) {
+            List<ResultSetFuture> futures = calculateReadTimeBuckets(query.getStartTime(), query.getEndTime())
```

Comment:
This `if` branch looks very similar to [that one](https://github.com/zalando-zmon/kairosdb/pull/114/files#diff-5cc760fcd7185d42167a9ce3ff854de4R839-R846). Can we probably try to generalize it and extract to separate method, e.g. `collectFromRowTimeKeySplitIndex`?

Reply:
fixed